format-common/src/main/java/co/cask/hydrator/format/input/TextInputProvider.java

```diff
@@ -28,42 +29,80 @@
+    if (offsetField != null) {
+      Schema offsetSchema = offsetField.getSchema();
+      Schema.Type offsetType = offsetSchema.isNullable() ? offsetSchema.getNonNullable().getType() :
+        offsetSchema.getType();
+      if (offsetType != Schema.Type.LONG) {
+        throw new IllegalArgumentException(String.format("The 'offset' field must be of type 'long', but found '%s'",
+                                                         offsetType.name().toLowerCase()));
       }
     }
 
-    if (schema == null) {
-      schema = getDefaultSchema(properties.get("pathField"));
+    Schema.Field bodyField = schema.getField("body");
+    if (bodyField == null) {
+      throw new IllegalArgumentException("The schema for the 'text' format must have a field named 'body'");
     }
+    Schema bodySchema = bodyField.getSchema();
+    Schema.Type bodyType = bodySchema.isNullable() ? bodySchema.getNonNullable().getType() : bodySchema.getType();
+    if (bodyType != Schema.Type.STRING) {
+      throw new IllegalArgumentException(String.format("The 'body' field must be of type 'string', but found '%s'",
+                                                       bodyType.name().toLowerCase()));
+    }
+
+    // fields should be body (required), offset (optional), [pathfield] (optional)
+    boolean expectOffset = schema.getField("offset") != null;
+    boolean expectPath = pathField != null;
+    int numExpectedFields = 1;
+    if (expectOffset) {
+      numExpectedFields++;
+    }
+    if (expectPath) {
+      numExpectedFields++;
+    }
+    int maxExpectedFields = pathField == null ? 2 : 3;
+    int numFields = schema.getFields().size();
+    if (numFields > numExpectedFields) {
+      String expectedFields;
+      if (expectOffset && expectPath) {
+        expectedFields = String.format("'offset', 'body', and '%s' fields", pathField);
+      } else if (expectOffset) {
+        expectedFields = "'offset' and 'body' fields";
+      } else if (expectPath) {
+        expectedFields = String.format("'body' and '%s' fields", pathField);
+      } else {
+        expectedFields = "'body' field";
+      }
+
+      int numExtraFields = numFields - maxExpectedFields;
+      throw new IllegalArgumentException(
+        String.format("The schema for the 'text' format must only contain the %s, but found %d other field%s",
+                      expectedFields, numExtraFields, numExtraFields > 1 ? "s" : ""));
+    }
+
     return new TextInputFormatter(schema);
   }
 
   public static Schema getDefaultSchema(@Nullable String pathField) {
     List<Schema.Field> fields = new ArrayList<>();
     fields.add(Schema.Field.of("offset", Schema.of(Schema.Type.LONG)));
-    fields.add(Schema.Field.of("body", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
-    if (pathField != null) {
-      fields.add(Schema.Field.of(pathField, Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    fields.add(Schema.Field.of("body", Schema.of(Schema.Type.STRING)));
```

Comment:
Is there a reason we change this to not nullable?

Reply:
it wasn't nullable in the previous release, and the widget for the file starts off with it not nullable.

It only became nullable during some recent refactoring, and reverting it back